back/spec/policies/initiative_policy_spec.rb

```diff
@@ -0,0 +1,300 @@
+            it { is_expected.not_to permit(:create) }
+            it { is_expected.not_to permit(:update)  }
+            it { is_expected.not_to permit(:destroy) }
+
+            it 'indexes the initiative' do
+              expect(scope.resolve.size).to eq 1
+            end
+          end
+
+          context 'for a visitor' do
+            let(:user) { nil }
+
+            it { is_expected.to permit(:show) }
+            it { is_expected.to permit(:by_slug) }
+            it { expect { policy.create? }.to raise_error(Pundit::NotAuthorizedError) }
+            it { expect { policy.update? }.to raise_error(Pundit::NotAuthorizedError) }
+            it { expect { policy.destroy? }.to raise_error(Pundit::NotAuthorizedError) }
+
+            it 'indexes the initiative' do
+              expect(scope.resolve.size).to eq 1
+            end
+          end
+        end
+      end
+    end
+
+    # For statuses with any CODES:
+    context "for initiatives with any status code (except 'custom')" do
+      (InitiativeStatus::CODES - ['custom']).each do |code|
+        context "for an initiative with #{code} status" do
+          let!(:initiative) do
+            create(:initiative, author: author, initiative_status: create("initiative_status_#{code}".to_sym))
+          end
+
+          context 'for an admin' do
+            let(:user) { create(:admin) }
+
+            it { is_expected.to permit(:show) }
+            it { is_expected.to permit(:by_slug) }
+            it { is_expected.to permit(:create) }
+            it { is_expected.to permit(:update)  }
+            it { is_expected.to permit(:destroy) }
+
+            it 'indexes the initiative' do
+              expect(scope.resolve.size).to eq 1
+            end
+          end
+
+          # Author can edit/delete initiative if approval feature not active, regardless of status.
+          context 'for a user who is author of the initiative' do
+            let(:user) { author }
+
+            it { is_expected.to permit(:show) }
+            it { is_expected.to permit(:by_slug) }
+            it { is_expected.to permit(:create) }
+            it { is_expected.to permit(:update)  }
+            it { is_expected.to permit(:destroy) }
+
+            it 'indexes the initiative' do
+              expect(scope.resolve.size).to eq 1
+            end
+          end
+        end
+      end
+    end
+  end
+
+  # ------------------- 2. Approval feature IS active ----------------------
+
+  context 'when approval feature IS fully active' do
+    before do
+      SettingsService.new.activate_feature! 'initiative_approval'
+
+      configuration = AppConfiguration.instance
+      configuration.settings['initiatives'] = {
+        enabled: true,
+        allowed: true,
+        require_approval: true, # This is also required to activate the feature
+        reacting_threshold: 2,
+        days_limit: 20,
+        threshold_reached_message: { 'en' => 'Threshold reached' },
```

Comment:
Probably, you don't need to update all initiative settings, just updating `require_approval` could be enough. Like here https://github.com/CitizenLabDotCo/citizenlab/pull/5416/files#diff-50c3d7a61dbe662b5566b5e47dae629f27dd165df14689e746ee0f5a6db940adR49

Reply:
```
Failure/Error: configuration.save!
       
       ActiveRecord::RecordInvalid:
         Validation failed: Settings {:fragment=>"#/initiatives", :error=>"RequiredSettings", :human_message=>"The active feature '#/initiatives' did not have all the required settings [\"reacting_threshold\", \"days_limit\", \"threshold_reached_message\", \"eligibility_criteria\"] in schema 45234ffe-3db1-5e8e-8d35-31b0c375820e"}
```